analytics.go

```diff
@@ -250,25 +251,25 @@ func (c *client) upload(b []byte) error {
 	}
 
 	defer res.Body.Close()
-	c.report(res)
-
-	return nil
+	return c.report(res)
 }
 
 // Report on response body.
-func (c *client) report(res *http.Response) {
+func (c *client) report(res *http.Response) (err error) {
+	var body []byte
+
 	if res.StatusCode < 400 {
```

Comment:
OH I also realized our Api doesn't return 200 always (https://github.com/segmentio/tracking-api/blob/9b18d72c74944369243da16a12e255b959640b19/server/server.go#L65-L71), it will return  a 400 if your json is invalid.